scripts/directtools/__init__.py

```diff
@@ -0,0 +1,450 @@
+from __future__ import (absolute_import, division, print_function)
+
+import collections
+from mantid import mtd
+from mantid.simpleapi import DeleteWorkspace, LineProfile, OneMinusExponentialCor, Transpose
+import matplotlib
+from matplotlib import pyplot
+import numpy
+from scipy import constants
+import time
+
+
+def _chooseMarker(markers, index):
+    """Pick a marker from markers and return next cyclic index."""
+    if index >= len(markers):
+        index = 0
+    marker = markers[index]
+    index += 1
+    return marker, index
+
+
+def _clearlatex(s):
+    """Return a string from which LaTeX formatting has been removed."""
+    s = s.translate(None, '$\\^{}')
+    s = s.replace('$', '')
+    s = s.replace('\\', '')
+    s = s.replace('^', '')
+    s = s.replace('{', '')
+    s = s.replace('}', '')
+    return s
+
+
+def _finalizeprofileE(axes):
+    """Set axes for const E axes."""
+    axes.set_xlim(xmin=0.)
+    axes.set_xlabel('$Q$ (\\AA$^{-1}$)')
+    axes.set_ylim(0.)
+    xMin, xMax = axes.get_xlim()
+    print('Auto Q-range: {}...{} \xc5-1'.format(xMin, xMax))
+
+
+def _finalizeprofileQ(workspaces, axes):
+    """Set axes for const Q axes."""
+    workspaces = _normwslist(workspaces)
+    axes.set_xlim(xmin=-10.)
+    axes.set_xlabel('Energy (meV)')
+    cMax = 0.
+    for i in range(len(workspaces)):
+        c = numpy.nanmax(workspaces[0].readY(0))
```

Comment:
Is this loop doing what you want? I might be missing something, but would you not get `cMax` with:
```
for ws in workspaces:
    c = numpy.nanmax(ws.readY(0))
    blah...
```